src/scripts/quests/QuestHelper.ts

```diff
@@ -135,16 +135,28 @@ class QuestHelper{
         return Math.floor(n + 1);
     }
 
-    public static haveFreeQuestSlots(): boolean {
-        return player.currentQuests().length < this.questSlots()();
+    public static canStartNewQuest(): boolean {
+        // Two conditions for starting new quest:
+        // 1. Current quests not exceed maximum slots
+        // 2. At least one quest is neither completed nor in-progress
+        if (player.currentQuests().length >= this.questSlots()()) {
+            return false;
+        }
+        for (let i = 0; i < QuestHelper.questList().length; i++) {
+            if (!(player.completedQuestList[i]() || QuestHelper.questList()[i].isCompleted() 
+                    || QuestHelper.questList()[i].inProgress()())) {
+                return true;
```

Comment:
I _think_ you could replace the whole `for` block with this?
```javascript
return QuestHelper.questList().some(q=>!q.isCompleted() && !q.inProgress()());
```

Reply:
You do need to check against `player.completedQuestList[i]()`, because this stores completed quests from previous game sessions, while `Quest.isCompleted()` only gets set to true when you complete the quest in the current session.